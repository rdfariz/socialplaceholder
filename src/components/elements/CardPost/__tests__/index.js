import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Card from '../';
import { shallow } from 'enzyme';

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useHistory: () => ({
    push: jest.fn(),
  }),
}));

describe('src/components/elements/CardPost', () => {
  const props = {
    title: 'title',
    id: '1',
    body: 'tesbody',
    userId: '10'
  };

  test('render', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(
      <Card {...props} isDetail />
    );
    expect(tree).toMatchSnapshot();
  });
  
  test('should calling function edit when button edit clicked', () => {
    const onDelete = jest.fn();
    const wrapper = shallow(<Card {...props} handleDeletePost={onDelete} isDetail />);
    const btnDelete = wrapper.find('.btn-delete');
    btnDelete.first().simulate('click');
    expect(onDelete).toBeCalled();
  });

  test('should return handleClick when card clicked', () => {
    const wrapper = shallow(<Card {...props} />);
    const card = wrapper.find('.card_post');
    if (card.exists()) {
      card.first().simulate('click');
    }
  });
  
  test('should return handleClick when card detail clicked', () => {
    const wrapper = shallow(<Card {...props} isDetail />);
    const card = wrapper.find('.card_post');
    if (card.exists()) {
      card.first().simulate('click');
    }
  });
});
