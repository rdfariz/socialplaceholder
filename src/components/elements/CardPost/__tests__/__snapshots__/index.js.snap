// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/elements/CardPost render 1`] = `
<div
  className=" flex"
>
  <div
    className="card_post w-full p-3 md:p-5 bg-white rounded-lg border false"
    onClick={[Function]}
  >
    <div>
      <div
        className="text-blue-600 font-medium"
      >
        <Link
          to="/users/10"
        >
          Author ID: 
          10
        </Link>
        <hr
          className="my-3"
        />
      </div>
      <h2
        className="text-gray-800 text-lg font-semibold false"
      >
        title
      </h2>
      <p
        className="mt-2 text-sm text-gray-600"
      >
        tesbody
      </p>
      <div
        className="mt-4"
      >
        <button
          className="btn-edit bg-blue-400 text-white font-bold py-2 px-4 rounded mr-2"
          onClick={[Function]}
        >
          Edit
        </button>
        <button
          className="btn-delete bg-pink-500 text-white font-bold py-2 px-4 rounded"
          onClick={[Function]}
        >
          Delete
        </button>
      </div>
    </div>
  </div>
</div>
`;
